packages/typescript-estree/src/node-utils.ts

```diff
@@ -649,16 +649,26 @@ export function convertTokens(ast: ts.SourceFile): TSESTree.Token[] {
   return result;
 }
 
-export interface TSError {
-  index: number;
-  lineNumber: number;
-  column: number;
-  message: string;
+export class TSError extends Error {
+  constructor(
+    message: string,
+    public readonly fileName: string,
+    public readonly index: number,
+    public readonly lineNumber: number,
```

Comment:
i think we should rename this property, lineNumber is a field present already in Error
```ts
Object.getOwnPropertyNames(new Error())
// [ "fileName", "lineNumber", "columnNumber" ]
```

Reply:
what do you think about changing it to

```ts
export interface TSErrorDetails {
  fileName: string;
  index: number;
  lineNumber: number;
  columnNumber: number;
}

export class TSError extends Error {
  constructor(message: string, public readonly details: TSErrorDetails) {
    super(message);
    Object.defineProperty(this, 'name', {
      value: new.target.name,
      enumerable: false,
      configurable: true,
    });
  }
}
```